pkg/ddl/ddl.go

```diff
@@ -1017,6 +1099,42 @@ func setDDLJobQuery(ctx sessionctx.Context, job *model.Job) {
 	}
 }
 
+func setDDLJobVersion(job *model.Job) {
+	if variable.DDLVersion.Load() != model.TiDBDDLV2 {
+		job.Version = model.TiDBDDLV1
```

Comment:
`job.Version` should not be used like this. I think the semantic of `job.Version` is similar to bootstrap version, the latest TiDB should always use the latest version number, and new tidb can handle old job version.

I suggest adding a new field `DistExec bool`(or `LocalMode bool`) to `model.Job` to achieve the same purpose here.